### PR TITLE
new addon: `fave-tab`

### LINKED ISSUE #4057: Favourites Tab
### Is your request related to a problem? Please describe.

Scratch's favourite system doesn't make it very easy to find a specific favourited project if there are a lot of them.

### Describe the solution you'd like

An extension pop-up tab that lists all of the user's favourite projects with a search bar for easily finding one.

### Describe alternatives you've considered

Using Cloud Games once it's customizable, but it takes up more space because it shows the online players and isn't searchable.

### Additional context

Maybe there could also be a toggle to show someone else's favourites when visiting his profile.

### PR DESCRIPTION
**Pull request by Samq64**
Resolves #4057

### Changes

Adds a new extension popup addon which shows the currently logged user's favorite projects.

### Reason for changes

It makes opening favorite projects easier.

### Tests

Tested on Chromium 103. Still WIP, but it works.

### CODE REVIEW

File: popups/fave-tab/popup.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export default async ({ addon, msg, safeMsg }) => {
+  window.vue = new Vue({
+    el: "body",
+    data: {
+      projects: [],
+      projectsVisable: false,
+    },
+    async created() {
+      const username = await addon.auth.fetchUsername();
+      const res = await fetch(`https://api.scratch.mit.edu/users/${username}/favorites/?limit=40`);
```
Comment (Samq64) [inline popups/fave-tab/popup.js:10]: Is there a way to fetch more than 40 projects? I want to add a search bar to and be able to search all favorites.

Comment (lisa-wolfgang) [reply popups/fave-tab/popup.js:10]: I believe that there's a `page` parameter for fetching the next set of 40, but I haven't checked if that's implemented for favorites.

**Comment by WorldLanguages** [comment]
I think some languages don't have enough space for 4 tabs on the popup 🤔

**Comment by lisa-wolfgang** [comment]
Can the popup be dynamically made wider?

**Comment by lisa-wolfgang** [comment]
Another option would be to collapse unopened tabs by hiding the text.

File: popups/fave-tab/popup.js (hunk 1 above)
Comment (CST1229) [reply popups/fave-tab/popup.js:10]: There's an `offset` parameter that offsets individual items, so you can set it to multiples of 40 to show different pages.

**Comment by mxmou** [comment]
> Another option would be to collapse unopened tabs by hiding the text.

I think we should do that, but only when there isn't enough space to display them normally.

**Comment by WorldLanguages** [comment]
We're not ready yet to support a third popup addon, so we should fix that if we consider this popup is worth adding.

@Samq64 Can you explain the motivation for making this a popup addon? And I just searched over sent feedback, and a common suggestion is to add a search bar to favorite projects, but it could also be added to /users/x/favorites instead of inside this new popup.

**Comment by Samq64** [comment]
> @Samq64 Can you explain the motivation for making this a popup addon? And I just searched over sent feedback, and a common suggestion is to add a search bar to favorite projects, but it could also be added to /users/x/favorites instead of inside this new popup.

You're right, that would work just as well. I guess I'll do that instead, but won't make a PR just yet.

**Comment by Secret-chest** [comment]
Why? The popup tab had easy access! @WorldLanguages

**Comment by Samq64** [comment]
I'm going to re-open this and if a search bar is added on the favorites page, this pr can me linked to be closed.

**Comment by lisa-wolfgang** [comment]
I think W_L is concerned that regular browser bookmarks achieve the same purpose that a popup tab would. However, I would disagree -- realistically, most users don't have all of their favorites bookmarked. In addition, some users may have thousands of favorited projects. This would make it convenient for the user to access their favorite projects without navigating to their profile's favorites page every time, even if they have that specific page bookmarked. There could even be another feature added to justify the tab's existence beyond convenience, like a very simple recommendation system that cycles through the list randomly.

**Pull request closed without merging**